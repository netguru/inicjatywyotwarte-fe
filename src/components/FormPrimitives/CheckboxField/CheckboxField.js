import React from 'react'
import styled from 'styled-components'
import Label from 'components/FormPrimitives/FieldLabel/FieldLabel'

const Checkbox = styled.div`
  display: flex;
  align-items: center;
`
Checkbox.displayName = 'Checkbox';

const Input = styled.input`
  width: 1rem;
  height: 1rem;
  margin-left: 0;
`
Input.displayName = 'Input';

const LabelCheckbox = styled(Label)`
  margin-right: 2rem;
  height: 1rem;
  display: flex;
  align-items: center;
  cursor: pointer;

  @media (max-width: 400px) {
    margin-right: 1.4rem;
  }
`
LabelCheckbox.displayName = 'LabelCheckbox';

export default function CheckboxField ({
  name,
  label
}) {
  return (
    <Checkbox>
      <Input type='checkbox' id={name} />
      <LabelCheckbox htmlFor={name}>{label}</LabelCheckbox>
    </Checkbox>
  )
}
