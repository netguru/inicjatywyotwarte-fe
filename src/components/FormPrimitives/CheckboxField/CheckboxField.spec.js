import React from 'react'
import CheckboxField from './CheckboxField'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CheckboxField', () => {
  test('renders div with proper class', () => {
    const wrapper = shallow(<CheckboxField/>);
    expect(wrapper.find('Checkbox').length).toEqual(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<CheckboxField/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


