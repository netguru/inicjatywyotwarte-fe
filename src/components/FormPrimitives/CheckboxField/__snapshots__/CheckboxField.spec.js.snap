// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField match snapshot 1`] = `
<Checkbox>
  <Input
    type="checkbox"
  />
  <LabelCheckbox />
</Checkbox>
`;
